core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -228,6 +235,25 @@ public Builder httpExecutor(HttpExecutor val) {
             this.httpExecutor = val;
             return this;
         }
+        public Map<String, String> buildBaseHeaders() {
+            if (Boolean.TRUE.equals(preemptiveBasicAuth)) {
+                Credentials creds = credentials;
+                if (creds == null) {
+                    throw new IllegalArgumentException("Must not enable preemptiveBasicAuth when there are no credentials, in feed for "+baseUri);
```

Comment:
I tend to find the more-situation-specific Exceptions most useful from the developer point of view, as it can be handy to add breakpoints on specific types when debugging; if you want to break when the exception is thrown it can sometimes be a bother to set it on the more generic type, as it may get thrown in other places.  

I agree there's no compelling reason to change the `IllegalArgumentException` here, however.  A separate discussion could be had about possible more useful base exceptions.  `IllegalBrooklynConfigurationException`, anyone? :-)